analytics/upress_reporting/helpers/download_data_aggregator.py

```diff
@@ -114,25 +117,73 @@ def _match_log_info_with_frbr_data(self, log_object):
+            item, to the edition, to the work, and to the original source record.
+
+            Links are linked to items by the item_id. 
+            Items are linked to editions by the edition_id. 
+            Editions are linked to works by the work_id.
+            Editions are linked to records by the dcdw_uuids.
+            '''
+
             for item in self.db_manager.session.query(Item).filter(
-                Item.source == self.publisher
-            ):
+                    Item.source == self.publisher):
                 for link in (
                     self.db_manager.session.query(Link)
                     .join(ITEM_LINKS)
                     .filter(ITEM_LINKS.c.item_id == item.id)
                     .filter(Link.media_type == "application/pdf")
                     .filter(Link.url.contains(link_group.strip()))
-                    .all()
-                ):
-                    item_edit_id = item.edition_id
+                        .all()):
                     for edit in self.db_manager.session.query(Edition).filter(
-                        Edition.id == item_edit_id
-                    ):
+                            Edition.id == item.edition_id):
                         title_parse = edit.title
-                        id_parse = edit.id
-
-            return [title_parse, match_time.group(0), id_parse]
+                        copyright_year = self._pull_copyright_year(edit)
+
+                        for record in self.db_manager.session.query(Record).filter(
+                                Record.uuid.in_(edit.dcdw_uuids)):
+                            book_id = (record.source_id).split("|")[0]
+                            usage_type = self._determine_usage(record)
+                            isbn = None
+
+                        for work in self.db_manager.session.query(Work).filter(
+                                Work.id == edit.work_id):
+                            authors = [author["name"]
+                                       for author in work.authors]
+                            disciplines = [subject["heading"]
+                                           for subject in work.subjects]
+                    
+                    # TODO: there are multiple ISBNs. where do we find eisbn?
+
+            return InteractionEvent(
+                title=title_parse,
+                book_id=book_id,
+                authors=authors,
+                isbn=None,
+                eisbn=None,
+                copyright_year=copyright_year,
+                disciplines=disciplines,
+                usage_type=usage_type,
+                interaction_type=InteractionType.DOWNLOAD,
+                timestamp=match_time.group(0)
+            )
+
+    def _pull_copyright_year(self, edition):
+        for date in edition.dates:
+            if "copyright" in date["type"]:
+                return date["date"]
+        # TODO: what if copyright year doesn't exist?
+        return None
```

Comment:
For readability, I like to have a newline before my return statement. Add a visible delimiter between what's going in the function and what the function returns.